File: src/test/scala/com/johnsnowlabs/nlp/annotators/er/EntityRulerTestSpec.scala
package com.johnsnowlabs.nlp.annotators.er

import com.johnsnowlabs.nlp.annotators.SparkSessionTest
import org.apache.spark.ml.Pipeline
import org.scalatest.flatspec.AnyFlatSpec

class EntityRulerTestSpec extends AnyFlatSpec with SparkSessionTest {

  "Entity Ruler" should "serialize" in {
    import spark.implicits._

    val textDataSet = Seq("John Snow is a good boss").toDS.toDF("text")
    val entityRuler = new EntityRuler()
      .setInputCols("document", "token")
      .setOutputCol("entities")

    val pipeline = new Pipeline().setStages(Array(documentAssembler, tokenizer, entityRuler))

    val entityRulerModel = pipeline.fit(emptyDataSet)
    val resultDataSet = entityRulerModel.transform(textDataSet)
    resultDataSet.show()
    println("")
  }

}

File: python/test/annotator/er/entity_ruler_test_spec.py
class EntityRulerTestSpec(unittest.TestCase):

    def setUp(self):
        self.data = SparkContextForTest.spark.createDataFrame([["John Snow lives in Winterfell"]]).toDF("text")
        self.path = os.getcwd() + "/../src/test/resources/entity-ruler/patterns.json"

    def runTest(self):
        document_assembler = DocumentAssembler().setInputCol("text").setOutputCol("document")
        tokenizer = Tokenizer().setInputCols("document").setOutputCol("token")

        entity_ruler = EntityRulerApproach() \
            .setInputCols(["document", "token"]) \
            .setOutputCol("entity") \
            .setPatternsResource(self.path)

        pipeline = Pipeline(stages=[document_assembler, tokenizer, entity_ruler])
        model = pipeline.fit(self.data)
        model.transform(self.data).show()

